spec/models/miq_expression_spec.rb

```diff
@@ -1631,4 +1631,138 @@
+        end
+
+        it "returns false if field belongs to 'has_and_belongs_to_many' association" do
+          field = "ManageIQ::Providers::InfraManager::Vm.storages-name"
+          expression = {"CONTAINS" => {"field" => field, "value" => "abc"}}
+          expect(described_class.new(nil).sql_supports_atom?(expression)).to eq(false)
+        end
+
+        it "returns false if field belongs to 'has_many' polimorhic/polimorhic association" do
+          field = "ManageIQ::Providers::InfraManager::Vm.advanced_settings-region_number"
+          expression = {"CONTAINS" => {"field" => field, "value" => "1"}}
+          expect(described_class.new(nil).sql_supports_atom?(expression)).to eq(false)
+        end
+
+        it "returns true if field belongs to 'has_many' association" do
+          # delegation to #field_in_sql?
+          field = "ManageIQ::Providers::InfraManager::Vm.registry_items-name"
+          expression = {"CONTAINS" => {"field" => field, "value" => "abc"}}
+          expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(true)
+        end
+      end
+    end
+
+    context "expression key is 'INCLUDE'" do
+      it "returns false for model-virtualfield" do
+        # delegation to #field_in_sql?
+        field = "ManageIQ::Providers::InfraManager::Vm-v_datastore_path"
+        expression = {"INCLUDES" => {"field" => field, "value" => "abc"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(false)
+      end
+
+      it "returns true for model-field" do
+        # delegation to #field_in_sql?
+        field = "ManageIQ::Providers::InfraManager::Vm-location"
+        expression = {"INCLUDES" => {"field" => field, "value" => "abc"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(true)
+      end
+
+      it "returns false for model.association.virtualfield" do
+        # delegation to #field_in_sql?
+        field = "ManageIQ::Providers::InfraManager::Vm.ext_management_system-hostname"
+        expression = {"INCLUDES" => {"field" => field, "value" => "abc"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(false)
+      end
+
+      it "returns true for model.accociation.field" do
+        # delegation to #field_in_sql?
+        field = "ManageIQ::Providers::InfraManager::Vm.ext_management_system-name"
+        expression = {"INCLUDES" => {"field" => field, "value" => "abc"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(true)
+      end
+
+      it "returns false if format of field is model.association..association-field" do
+        field = "ManageIQ::Providers::InfraManager::Vm.service.miq_request-v_approved_by"
+        expression = {"INCLUDES" => {"field" => field, "value" => "abc"}}
+        expect(described_class.new(expression).sql_supports_atom?(expression)).to eq(false)
+      end
+    end
+
+    context "not supported expression keys" do
```

Comment:
I don't this it is necessary for these tests to be in a separate context. I would just remove the context and leave the cases.